### DISCUSSION TITLE
CSS/SCSS/SASS modules class type safety with TS when importing to .tsx files

### DISCUSSION THREAD

**Discussion by TobiasKullblikk** [solidjs/solid#1983] [Ideas]
Given the example below, it would be really nice with type safety when importing CSS modules to your .tsx files. So if I rename a CSS class, the corresponding variable will be updated in all files where it is used. Errors if using a non-existent class as well. 

Maybe this is better suited as a VS Code/extension feature request?

```
import s from './InputBox.module.scss'

const classList = { [s.inputBox]: true };
```

**Comment by lxsmnsyc** [comment]
This is unrelated to SolidJS, and I believe the solution for this already exists.

https://www.npmjs.com/package/typed-css-modules
